pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -272,3 +275,55 @@ func (k *Kaniko) getKanikoJobSpec(namespace string, buildOptions *BuildOptions,
 
 	return kanikoJobSpec
 }
+
+func (k *Kaniko) getJobLogs(namespace string, jobName string) (string, error) {
+	k.logger.DebugWith("Fetching kaniko logs", "namespace", namespace, "job", jobName)
+
+	// list pods
+	jobPods, err := k.kubeClientSet.CoreV1().Pods(namespace).List(meta_v1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", jobName),
+	})
+
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to list job's pods")
+	}
+	if len(jobPods.Items) == 0 {
+		return "", errors.Wrapf(err, "No pods found for job")
+	}
+	if len(jobPods.Items) > 1 {
+		return "", errors.Wrapf(err, "Got too many job pods")
+	}
+
+	// find job pod
+	restClientRequest := k.kubeClientSet.
+		CoreV1().
+		Pods(namespace).
+		GetLogs(jobPods.Items[0].Name, &v1.PodLogOptions{})
+
+	restReadCloser, err := restClientRequest.Stream()
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get log read/closer")
+	}
+
+	defer restReadCloser.Close() // nolint: errcheck
+
+	logContents, err := ioutil.ReadAll(restReadCloser)
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to read logs")
+	}
+
+	return string(logContents), nil
+}
+
+func (k *Kaniko) deleteJob(namespace string, jobName string) error {
+	k.logger.DebugWith("Deleting kaniko job", "namespace", namespace, "job", jobName)
+
+	propagationPolicy := meta_v1.DeletePropagationBackground
+	err := k.kubeClientSet.BatchV1().Jobs(namespace).Delete(jobName, &meta_v1.DeleteOptions{
```

Comment:
```suggestion
	if err := k.kubeClientSet.BatchV1().Jobs(namespace).Delete(jobName, &meta_v1.DeleteOptions{
		PropagationPolicy: &propagationPolicy,
	}); err != nil {
		return errors.Wrap(err, "Failed to delete kaniko job")
	}
	return nil
	
```

Reply:
Also,besides deleting the job, what about other resources related? are there any configmaps created for that kind of job? because that would be needed to be deleted as well

Reply:
The only additional resource I can think of is the tar file with build bundle but it's located in the temp folder and will be deleted automatically when temp folder is deleted